pkg/providers/instance/suite_test.go

```diff
@@ -291,4 +293,44 @@ var _ = Describe("InstanceProvider", func() {
 		Expect(len(interfaces)).To(Equal(1))
 		Expect(interfaces[0].Name).To(Equal(managedNic.Name))
 	})
+	Context("CSExtension", func() {
+		var originalOptions *options.Options
+
+		BeforeEach(func() {
+			originalOptions = options.FromContext(ctx)
+			ctx = options.ToContext(
+				ctx,
+				test.Options(test.OptionsFields{
+					ProvisionMode: lo.ToPtr(consts.ProvisionModeBootstrappingClient),
```

Comment:
since ProvisionMode is passed as a parameter to the InstanceProvider, we need to also recreate the azureEnv. 

This wont execute with CSE Extension, and when you get there I suspect you will run into an error about node bootstrapping URL not being able to find the host, since we dont implement a fake for it

Reply:
Removed tests overall, per discussion with Alex